### PR TITLE
docs(headless): range facets documentation

### PR DESCRIPTION
**Pull request by kaitlynkatz**
In this PR, I tried out not including any types for params and returns, and intellisense is still working correctly. Let me know if that approach makes sense.

However, I can't find where to document options so that they appear with intellisense- it doesn't seem to work the same way as the basic and category facets.

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]

**Comment by samisayegh** [comment]
> However, I can't find where to document options so that they appear with intellisense- it doesn't seem to work the same way as the basic and category facets.

Hi Kaitlyn, which options were you trying to document that did not show up with intellisense?

**Review by samisayegh — APPROVED** [review]

**Comment by kaitlynkatz** [comment]
> Hi Kaitlyn, which options were you trying to document that did not show up with intellisense?

All of them, actually... If I make a variable with the type `NumericFacetOptions`, there's no intellisense description on any of the properties.

File: packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-facet.ts [hunk 1]
```diff
@@ -55,15 +59,19 @@ export function buildDateFacet(
...
     /**
-     * Selects (deselects) the passed value if unselected (selected).
-     * @param selection The facet value to select or deselect.
+     * Toggles the specified facet value.
+     * @param selection The facet value to toggle.
      */
     toggleSelect: (selection: DateFacetValue) =>
       dispatch(executeToggleDateFacetSelect({facetId, selection})),
 
-    /** @returns The state of the `DateFacet` controller.*/
```
Comment (jpmarceau) [inline packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-facet.ts:64]: Why do we not need the `returns` here but we do at line 28 in the file above? (https://github.com/coveo/ui-kit/pull/376/files#diff-0aac8d5633bc178ffcc78e2b06d8fcadf8cd25c61d5aec7a2bfa150f7e2ff556R28)

Comment (jpmarceau) [reply packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-facet.ts:64]: ah nevermind, I didn't realize that above it was for options 👍

**Review by jpmarceau — APPROVED** [review]
GG!

**Pull request merged**